tests/ui/half-open-range-patterns/half-open-range-pats-thir-lower-empty.rs

```diff
@@ -9,36 +9,36 @@ macro_rules! m {
 
 fn main() {
     m!(0, ..u8::MIN);
-    //~^ ERROR lower range bound must be less than upper
+    //~^ ERROR upper bound for a range pattern cannot be the minimum
```

Comment:
Discussion: this is specifically applicable to the half-open range pattern, right? Because `..=u8::MIN`-the-closed-range would be fine. Maybe say

> ERROR: upper bound for a half-open range pattern cannot be the minimum

WDYT?

Reply:
I am not sure what a half-open range means, actually.. apparently `a..b` is called half-open? So I'd have to investigate the wording

Reply:
It refers to one of `[a, b)` or `(a, b]` where one of the ends are exclusive. In our case, `a..b` is the `[a, b)` right-side half-open range. However, if this phrase isn't familiar to you, then we should probably not use it since you won't be the only one unfamiliar with the phrase. (I suppose this is only a familiar wording given some discrete math courses or whatever, so yeah...)